io/disk/fiotest.py

```diff
@@ -73,6 +73,18 @@ def setUp(self):
         self.raid_create = False
         self.devdax_file = None
 
+        # Checking if self.disk input provided or not, if not Cancelling test
+        if self.disk:
+            # Checking if self.disk is a valid disk in OS
+            if self.disk in disk.get_disks():
+                #if self.dir is None using using self.workdir as self.dir
+                if self.dir is None:
+                    self.dir = self.workdir
```

Comment:
@AYUSHJAIN951 why you want to change dir only when the disk is given.
why cannot we take it out of this loop.
we already has a parameter in yaml , i,e dir. where the user will be providing the dir.
if you want to use the self.worldir, then use it in the following manner,

```
if not self.dir:
    self.dir = self.worldir
```
Please re-check on this.

Reply:
Because as you quoted in https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2560#:~:text=Note%20%3A%20These%20test%20cases%20are%20mainly%20intended%20for%20a%20block%20devices%20and%20may%20not%20be%20useful%20to%20run%20on%20only%20directory.
Note : These test cases are mainly intended for a block devices and may not be useful to run on only directory.

So if self.disk is not provided there is no point of changing self.dir anyways.

Reply:
@AYUSHJAIN951 I will re-iterate the same again here,

The test scripts were mainly for block devices as of now. if you want to extend the same script to run on any OS installed disk, following problems might occur,

1. since we are cleaning the disk before starting the test and after starting the test, your OS may get corrupt when you use dir from the OS.
2. few test cases consumes lot of space on the disk/dir and OS disk may not be capable of providing it all the time.

Your always welcome to add any new scenarios to these script, but make sure that your changes should not break the current flow or functional verification or should not lead to other problems.

take the example of the same fiotest here, we have pmem test available on this.
they have handled it using yaml and scripts in .py which is not disturbing other functionality verification here.

your always welcome to add the scenarios in that way.

 coming to #2560 , you were trying run the test on self.workdir and the test was cancelling when the disk is not given.
obviously when you want to run the test on self.workdir, user will not provide the disk value and test will cancel without running on either them, disk/dir.
That was a invalid scenario, and you might need to tweak it properly.

one more case,
if an user wants to run the test on some disk and he provides the disk in yaml, but due to some parsing issue, input parameters are not parsed and test ran on some temp dir and test pass.
then this is also a false PASS, as he wants to run the test on disk which was not happened but results say it is passed

So, as a summery, take care of all such cases and add any number of scenarios.

hope you got my view.